### ISSUE TITLE
Correct path structure for DOjS projects?

### ISSUE THREAD

**Issue by Axle-Ozz-i-sofT** [SuperIlu/DOjS#46]
Hi
Hi
Just testing out DOjS. This looks great :)

DOjS V1.10.0
Windows 95 OSR2.x, FreeDOS 1.3

I am testing DOjS examples and also attempting to test jiyuai V0.0.2
**Q. What are the script files path relative to?** (The script file location, or DOjS.EXE location)

DOjS script examples are a sub-directory of \DOjS aka \DOjS\examples\\\*.\*
Examples run without issue.

---------------
I placed jiyuai as a subdirectory \DOjS\jiyuai\\*.js
.\DOjS\jiyuai\MAIN.JS
.\DOjS\jiyuai\ui\ui.js
If I run the script from the script directory:
Batch
```
@ECHO OFF
SET PATH=%path%;C:\JSDOS\DOjS;C:\JSDOS\DOjS\jiyuai
REM CD C:\JSDOS\DOjS
CD C:\JSDOS\DOjS\jiyuai
CALL C:\JSDOS\DOjS\DOjS.EXE -r C:\JSDOS\DOjS\jiyuai\main.js
```

I get errors from: Include('./ui/uiapp.js')
ReferenceError: 'Include" is not defined
DOjS ERROR

---------------
If i run the .\DOjS\jiyuai\main.js script from the DOjS editor.
`SET PATH=%path%;C:\JSDOS\DOjS;C:\JSDOS\DOjS\jiyuai`
I get:
```
Erro: Could not load "./ui/uiapp.js"
    at Require (JSBOOT.ZIP=JSBOO/func.js:214)
    at Include (JSBOOT.ZIP=JSBOO/func.js:224)
    at C:\JSDOS\DOjS\jiyuai\main.js:4
```

main.js
```
var inputStr = '';
Include('./ui/uiapp.js');
Include('./demos/dlgdemo.js');
Include('./demos/scrdemo.js');
Include('./demos/ascdemo.js');
Include('./demos/wdgdemo.js');

...
```
I have tried changing the paths to reflect the location of the script in relation to DOjS directory, ie. Include('./jiyuai/ui/uiapp.js'); Include('jiyuai/ui/uiapp.js');
I have some partial success using DOS file names: Include('.\ui\uiapp.js');

-----------------
If I place jiyuai files in the root directory next to DOjS everything runs as expected **without** error.
```
\DOjS\DOjS.EXE
\DOjS\main.js (jiyuai)
\DOjS\ui\\*.js
\DOjS\ui\*.[DIR]
```
Must libraries be a sub-directory of DOjS.EXE?
and
Can I place the main script.js in a sub-directory?
Or are scripts meant to be in the root along side of DOjS.exe?

Sorry for the long explanation.
This may be related to jiyuai, but I also need to work out the correct location of scripts an paths for my self as well, otherwise I will end up with a root directory mess and name conflicts everywhere lol

Best regards
Axle

**Comment by SuperIlu** [comment]
Hi,

all paths in DOjS are relative to the current directory when DOjS was started. There is a slight exception for JSBOOT ZIP or directory.

So if your current dir is `c:\dojs` and you run `dojs bar\myscript.js` the script is loaded from `c:\dojs\bar\myscript.js`.
If `myscript.js` uses `Include("foo");` a bunch of filenames are tried, but all relative to `c:\dojs`:
- `c:\dojs\foo`
- `c:\dojs\foo.js`
- `c:\dojs\jsboot.zip=foo`
- `c:\dojs\jsboot.zip=foo.js`
- `c:\dojs\jsboot\foo`
- `c:\dojs\jsboot\foo.js`

So much for the details.

So I guess your problem with jiyuai results in running DOjS from a directory above the `jiyuai` directory which leads to all includes below `ui` to be in the wrong place.

My recommendation for new projects would be:
- make a directory dedicated to your project.
- copy over DOjS.EXE, CWSDPMI.EXE, JSBOOT.ZIP and all DEXs you want to use
- place your main script in the same directory as DOjS
- put all your other scripts either in the same dir or into subdirectories
- update all Include() or Require() calls accordingly

I hope that helps a little?

**Comment by Axle-Ozz-i-sofT** [comment]
Hi [SuperIlu](https://github.com/SuperIlu)
Sorry for the late reply from my question.

Thank you very much for confirming this. It's nice to walk into an illuminated room :)
I have used VBScript and JScript in the past with MS.hta so I find this very interesting.
I also created some apps way back with RFO BASIC on android witch uses a JavaScript call back and headless browser.

I am just looking at what graphics library I am going to use with DJGPP and comparing that with DOjS. If I can create a simple GUI wrapper for DOjS I think I will have a lot of fun :)

Again thank you very much.
Axle
